WSDiscovery.py

```diff
@@ -1,6 +1,16 @@
 #!/usr/bin/env python
 
-import urllib.request, urllib.parse, urllib.error
+import sys
+
+if sys.version_info.major > 2:
```

Comment:
Instead of this, do the urllib import in a `try:` block and fall back to urllib2 in your `except ImportError:`

This is the style suggested by https://docs.python.org/3/howto/pyporting.html#use-feature-detection-instead-of-version-detection . Since there's really very little that needs to be changed, doing a major version check is overkill IMO